src/TLuaInterpreter.cpp

```diff
@@ -6576,6 +6576,14 @@ int TLuaInterpreter::exists(lua_State* L)
         cnt += host.getAliasUnit()->mLookupTable.count(name);
     } else if (type == "keybind") {
         cnt += host.getKeyUnit()->mLookupTable.count(name);
+    } else if (type == "script") {
+        //Richard Moffitt: ugly hack. will fix it later for a pretty one-liner code...
+        std::list<TScript*> scripts = host.getScriptUnit()->getScriptRootNodeList();
+        for (auto script : scripts) {
+            if (!script->getScript().isEmpty()) {
```

Comment:
With the default content of a script being a comment, this is pretty unlikely to be empty.

Reply:
Is there any issue of scripts that can't be compiled - will that throw off the detection of functions - or isn't that what is being done here? :confounded:

Reply:
It's not an issue either way that would be to scripts specifically, and solving that would be leaking the scope of the PR. Feel free to investigate it anyway, but the fix should be in another PR to not to stall this one!

Reply:
okay!

Reply:
Okay - but the isEmpty() test still needs to go doesn't it?

Reply:
applied.